addons/l10n_fr_account/models/account_move.py

```diff
@@ -20,3 +28,11 @@ def _get_view(self, view_id=None, view_type='form', **options):
     def _compute_l10n_fr_is_company_french(self):
         for record in self:
             record.l10n_fr_is_company_french = record.country_code in record.company_id._get_france_country_codes()
+
+    def _post(self, soft=True):
+        # EXTENDS 'account'
+        res = super()._post(soft)
+        for move in self:
+            if move.l10n_fr_is_company_french and move.show_delivery_date:
```

Comment:
Doesn't really make sense that no ? you would override the value the user entered :sweat_smile: 
```suggestion
            if move.show_delivery_date and not move.delivery_date:
```